frontend/src/metabase/dashboard/containers/DashboardHeader.jsx

```diff
@@ -489,5 +524,8 @@ class DashboardHeader extends Component {
 
 export default _.compose(
   Bookmark.loadList(),
+  Collections.load({
+    id: (state, props) => props.dashboard.collection_id || "root",
+  }),
```

Comment:
I don't think we need this request (except that I guess we don't mock the type hydration in the entity loader) is this request because the root collection doesn't get hydrated?

Can't we just treat the collection prop as optional since the root collection will never be an instance analytics collection?

Reply:
In theory we could, but I'm not a fan of the fact that sometimes it's there and sometimes it isn't. It also doesn't look like we rely on that prop anywhere else in the app (collection isn't even on the dashboard type, for example), so I decided to get the collection from the source. I imagine most of the time this will simply get the collection from state, rather than require a network request

Reply:
I wonder if we could talk the backend team about making the root collection hydration more consistent? seems dumb to hydrate it most of the time, but for us to have to jump through conditional hoops on the front end because of it.

Reply:
Interesting idea about not having 2 sources of truth.

That sounds like a bug. It should be there always. Can you help me understand when it is not included? Is it only when it is the root collection?

I think there are 2 paths forward, make it work always, or make it work never. What are the benefits of having the contained-collection on a dashboard?

Reply:
Well, I think the main benefit is that there may be cases where it's 1 less request to make under certain circumstances. But I also think it could be the source of some bugs in the future unless the frontend updates it's collection entities accordingly when fetching a dashboard

Reply:
Yeah, loading the collections entity makes sense to me!